src/components/DishModal/DishModal.tsx

```diff
@@ -0,0 +1,157 @@
+    },
+    lineStyling: { width: "83.5px" },
+  };
+  return createPortal(
+    <div className={styles.modalOverlay} onClick={() => closeModalDesktop()}>
+      <div className={styles.DishModalLayout} onClick={(e) => e.stopPropagation()}>
+        <div className={styles.modalHeader}>
+          <button className={styles.btnContainer} onClick={() => closeModal()}>
+            {isMobileOrTablet ? <img src={blackXIcon} alt="black-x-icon" /> : <img src={whiteXIcon} alt="white-x-icon" />}
+          </button>
+        </div>
+        <div className={styles.heroContainer}>
+          <img src={dish.pictureUrl} alt={dish.name} />
+        </div>
+        <section className={styles.dishContentSection}>
+          <div className={styles.dishContentLayout}>
+            <div className={styles.dishContentContainer}>
+              <div className={styles.dishInformationContainer}>
+                {isMobileOrTablet ? (
+                  <DishCard
+                    dish={dish}
+                    key={dish.keyId}
+                    dishImageSize={mobileStyles.cardImage}
+                    cardWith={mobileStyles.cardWith}
+                    dishContentLayoutStyling={mobileStyles.cardContentLayout}
+                    dishTitleStyling={mobileStyles.cardTitle}
+                    dishDescriptionStyling={mobileStyles.cardDescription}
+                    shouldDisplayFoodIcon={mobileStyles.shouldDisplayFoodIcon}
+                    dishPriceContainerStyling={mobileStyles.dishPriceContainer}
+                  />
+                ) : (
+                  <DishCard
+                    dish={dish}
+                    key={dish.keyId}
+                    dishImageSize={desktopStyles.cardImage}
+                    cardWith={desktopStyles.cardWith}
+                    dishContentLayoutStyling={desktopStyles.cardContentLayout}
+                    dishTitleStyling={desktopStyles.cardTitle}
+                    dishDescriptionStyling={desktopStyles.cardDescription}
+                    shouldDisplayFoodIcon={true}
+                    dishPriceContainerStyling={desktopStyles.dishPriceContainer}
+                    dishContentContainerStyling={desktopStyles.cardContentContainer}
+                    dishPriceTextStyling={desktopStyles.dishPriceTextStyling}
+                    shouldDisplayLeftSideLine={true}
+                    shouldDisplayRightSideLine={true}
+                    lineStyling={desktopStyles.lineStyling}
+                  />
+                )}
+              </div>
+              <div className={styles.orderAddonsContainer}>
+                <ChooseSideDishes />
+                <ChooseDishChanges />
+                <ChooseQuantity />
+              </div>
+              <button className={styles.addToBagBtn}>ADD TO BAG</button>
+            </div>
+          </div>
+        </section>
+        {isMobileOrTablet && <Footer />}
+      </div>
+    </div>,
+    document.getElementById("modal")!
```

Comment:
This exclamation mark could result in a hard-to-find bug. What happens if for some reason this `#modal` element is not found? Please test that scenario, and handle (if needed) any resulting bug.

Reply:
@nitzanpap  done